br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -347,34 +337,34 @@ func (p *PreRestoreInfoGetterImpl) Init() {
 // GetAllTableStructures gets all the table structures with the information from both the source and the target.
 // It implements the PreRestoreInfoGetter interface.
 // It has a caching mechanism: the table structures will be obtained from the source only once.
-func (p *PreRestoreInfoGetterImpl) GetAllTableStructures(ctx context.Context) (map[string]*checkpoints.TidbDBInfo, error) {
+func (p *PreRestoreInfoGetterImpl) GetAllTableStructures(ctx context.Context, opts ...ropts.GetPreInfoOption) (map[string]*checkpoints.TidbDBInfo, error) {
 	var (
 		dbInfos map[string]*checkpoints.TidbDBInfo
 		err     error
 	)
-	dbInfosVal := ctx.Value(preInfoGetterKeyTableStructsCache)
-	if dbInfosVal != nil {
-		if v, ok := dbInfosVal.(map[string]*checkpoints.TidbDBInfo); ok {
-			dbInfos = v
-		}
+	getPreInfoCfg := p.getPreInfoCfg.Clone()
+	for _, o := range opts {
+		o(getPreInfoCfg)
 	}
-	if dbInfos != nil {
+	dbInfos = p.dbInfosCache
+	if dbInfos != nil && !getPreInfoCfg.ForceReloadCache {
```

Comment:
when is the `getPreInfoCfg.ForceReloadCache=fasle` used? I think we can remove this option.

Reply:
This is to provide a mechanism to fetch the info again rather than use the cached value.  In case some devs may use it.    Otherwise they cannot reload the value once the value is cached.

Reply:
Since it is an option, so it is backward compatible with the current code.  No need to do any adaptions for the devs.

Reply:
It has not been exposed to the Lightning user, just an internal calling. The reason I think we can remove it is that we may don't need to reload the value at all. It just cost the extra time for user.